File: C/sum of main diagonal.c
#include <stdio.h>
//This code simply makes the sum the main diagonal of a matrix.
void main(){
	//Declaring variables
	int tam, soma;
	//Scanning the size of the matrix.
	scanf("%d", &tam);
	/*The matrix uses the tam variable for it's size.
	  Example: If tam has the value of 3, the matrix will be a 3x3 matrix.*/
	int m[tam][tam];
	//Now, the user fills the matrix.
	for(int i = 0; i < tam; i++)
		for(int j = 0; j < tam; j++){
			scanf("%d", &m[i][j]);
		}
	//After that, the code will sum the main diagonal of the matrix.
	for(int i = 0; i < tam; i++)
		for(int j = 0; j < tam; j++){
			if (i == j)
				soma = soma + m[i][j];
		}
	//Next, for the final step, the program will print the results.
	printf("%d", soma);
}

File: JavaScript/sum of main diagonal.js
//This code simply makes the sum the main diagonal of a matrix.
//First we declare the variables we need.
var tam, principal, secondary;
principal = secondary = 0;
//After that, we ask the user the size of the matrix.
tam = parseInt(prompt("Type the matrix size (Example: if the value is 3, the program will consider it a 3x3 matrix):"));
var m = [];
//The following two for loops now adquire the matrix.
for(i = 0; i < tam; i++){
    m[i] = []
    for(j = 0; j < tam; j++){
        m[i][j] = parseInt(prompt(""));
    };
};
//Now, those two for loops sums the values of the main diagonal.
for(i = 0; i < tam; i++){
    for(j = 0; j < tam; j++){
        if (i == j){
            principal = principal + m[i][j];
        }
    };
};

for(i = 0; i < tam; i++){
    for(j = 0; j < tam; j++){
		if ((i + j) == (tam - 1)){
			secondary = secondary + m[i][j];
		}	
	};
};
	
//After that, the sum of the values is printed.
alert(`The sum of the principal diagonal is: ${principal}`);
alert(`The sum of the secondary diagonal is: ${secondary}`);
    